src/ccall.cpp

```diff
@@ -2085,6 +2094,11 @@ jl_cgval_t function_sig_t::emit_a_ccall(
         }
     }
 
+    Value *last_gc_state = NULL;
+    Value *ptls = get_current_ptls(ctx); // Do we need to reload this after the ccall?
+
+    if (gc_safe)
+        last_gc_state = emit_gc_safe_enter(ctx.builder, ctx.types().T_size, ptls, false);
```

Comment:
Since the OpBundle mutates gc-unsafe state locally, I think we actually will need to insert these later (after the gc-late-lowering pass) or at least teach that pass to accurately detect gc-safe/unsafe region transitions and move those region boundaries to inside of the runtime-interop code.

Reply:
Yeah I was getting the feeling that I would actually need to add GC state transition support to codegen